integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigrIstioGateway.java

```diff
@@ -0,0 +1,460 @@
+    String serverName = managedServerPrefix + "1";
+
+    // send a HTTP request to set http session state(count number) and save HTTP session info
+    // before shutting down the primary server
+    Map<String, String> httpDataInfo =
+        getServerAndSessionInfoAndVerify(serverName, webServiceSetUrl, " -c ");
+    // get server and session info from web service deployed on the cluster
+    String origPrimaryServerName = httpDataInfo.get(primaryServerAttr);
+    String origSecondaryServerName = httpDataInfo.get(secondaryServerAttr);
+    String origSessionCreateTime = httpDataInfo.get(sessionCreateTimeAttr);
+    logger.info("Got the primary server {0}, the secondary server {1} "
+        + "and session create time {2} before shutting down the primary server",
+        origPrimaryServerName, origSecondaryServerName, origSessionCreateTime);
+
+    // stop the primary server by changing ServerStartPolicy to NEVER and patching domain
+    logger.info("Shut down the primary server {0}", origPrimaryServerName);
+    shutdownServerUsingServerStartPolicy(origPrimaryServerName);
+
+    // send a HTTP request to get server and session info after shutting down the primary server
+    serverName = domainUid + "-" + origSecondaryServerName;
+    httpDataInfo = getServerAndSessionInfoAndVerify(serverName, webServiceGetUrl, " -b ");
+    // get server and session info from web service deployed on the cluster
+    String primaryServerName = httpDataInfo.get(primaryServerAttr);
+    String sessionCreateTime = httpDataInfo.get(sessionCreateTimeAttr);
+    String countStr = httpDataInfo.get(countAttr);
+    int count = Optional.ofNullable(countStr).map(Integer::valueOf).orElse(0);
+    logger.info("After patching the domain, the primary server changes to {0} "
+        + ", session create time {1} and session state {2}",
+        primaryServerName, sessionCreateTime, countStr);
+
+    // verify that a new primary server is picked and HTTP session state is migrated
+    assertAll("Check that WebLogic server and session vars is not null or empty",
+        () -> assertNotEquals(origPrimaryServerName, primaryServerName,
+            "After the primary server stopped, another server should become the new primary server"),
+        () -> assertEquals(origSessionCreateTime, sessionCreateTime,
+            "After the primary server stopped, HTTP session state should be migrated to the new primary server"),
+        () -> assertEquals(count, SESSION_STATE,
+            "After the primary server stopped, HTTP session state should be migrated to the new primary server")
+    );
+
+    finalPrimaryServerName = primaryServerName;
+
+    logger.info("Done testSessionMigration \nThe new primary server is {0}, it was {1}. "
+        + "\nThe session state was set to {2}, it is migrated to the new primary server.",
+        primaryServerName, origPrimaryServerName, SESSION_STATE);
+  }
+
+  private static int configIstioModelInImageDomain(String miiImage,
```

Comment:
We can move the method to MiiUtility class or overwrite an  existing method to take a boolean istioEnabled as extra parameter.

Reply:
changed

Reply:
This utility method has been moved to utils/SessionMigrationUtil.java instead of utils/CommonMiiTestUtils.java
Please move the method to utils/CommonMiiTestUtils.java

Reply:
moved configIstioModelInImageDomain to CommonMiiTestUtils.java